.github/maintainers_guide.md

```diff
@@ -15,45 +15,63 @@ $ brew update
 $ brew install pyenv
 ```
 
-Install necessary Python runtimes for development/testing. You can rely on Travis CI builds for testing with various major versions. https://github.com/slackapi/python-slack-sdk/blob/main/.travis.yml
+You can hook `pyenv` into your shell automatically by running `pyenv init` and following the instructions.
+
+Install necessary Python runtimes for development/testing. It is not necessary
+to install all the various Python versions we test in [continuous integration on
+GitHub Actions](https://github.com/slackapi/python-slack-sdk/blob/main/.github/workflows/ci-build.yml),
+but make sure you are running at least one version that we execute our tests in
+locally so that you can run the tests yourself.
 
 ```bash
 $ pyenv install -l | grep -v "-e[conda|stackless|pypy]"
 
-$ pyenv install 3.8.5 # select the latest patch version
-$ pyenv local 3.8.5
+$ pyenv install 3.9.6 # select the latest patch version
+$ pyenv local 3.9.6
 
 $ pyenv versions
   system
   3.6.10
   3.7.7
-* 3.8.5 (set by /path-to-python-slack-sdk/.python-version)
+* 3.9.6 (set by /path-to-python-slack-sdk/.python-version)
 
 $ pyenv rehash
 ```
 
-Then, you can create a new Virtual Environment this way:
+Then, you can create a new [Virtual Environment](https://docs.python.org/3/tutorial/venv.html) specific to the Python version you just installed by running:
 
 ```
-$ python -m venv env_3.8.5
-$ source env_3.8.5/bin/activate
+$ python -m venv env_3.9.6
+$ source env_3.9.6/bin/activate
 ```
 
+At this point you have a clean, Python-version-specific environment "activated" for
+use just for this project. All `python` and `pip` commands run in your shell
+from this point on run in the context of this virtual environment. You can
+deactivate the virtual environment by running `deactivate`; it is recommended to
+do so after you are done working in this project. To come back to development
+work for this project again in the future, `cd` into this project directory and
+run `source env_3.9.6/bin/activate` again.
+
+The last step is to install this project's dependencies; to do so, check out [how
+we configure GitHub Actions to install dependencies for this project for use in
+our continuous integration](https://github.com/slackapi/python-slack-sdk/blob/main/.github/workflows/ci-build.yml#L26-L30). You can also run `./scripts/run_validation.sh` to install the dependencies and run the unit tests in one command!
+
 ## Tasks
 
 ### Testing (Unit Tests)
 
 When you make changes to this SDK, please write unit tests verifying if the changes work as you expected. You can easily run all the tests by running the command. The `validate` command runs Flake8 (static code analyzer), Black (code formatter), and unit tests in the `tests` directory for you.
 
 ```bash
-python setup.py validate # run all
+PYTHON_SLACK_SDK_MOCK_SERVER_MODE="threading" CI_UNSTABLE_TESTS_SKIP_ENABLED="1" python setup.py validate # run all
```

Comment:
Did you pull this from the github action?

Reply:
Indeed

Reply:
Without these env vars the tests fail for me locally

Reply:
What error do you face? `CI_UNSTABLE_TESTS_SKIP_ENABLED` is a workaround only for GitHub Actions env (I was not able to find any solution for it at the time). I don't recommend enabling this for local development. If the error is 100% reproducible on your local env, we should fix the issue.

Reply:
OK, I will revert this change in the docs then. Should I file a separate issue for test failures? The short summary is that the tests remain stuck at this point of the output if I run without these env vars: 

```
tests/slack_sdk/socket_mode/test_interactions_builtin.py .F                       [ 47%]
tests/slack_sdk/socket_mode/test_interactions_websocket_client.py
```

Reply:
I can run the tests individually / separately per file to collect the log output, too, but perhaps a separate issue would be a better venue for that.

Reply:
@filmaj For the failure, if it's the same with https://github.com/slackapi/python-slack-sdk/issues/808, we can reopen it (I know the reset by peer errors still occasionally occur). For the stuck test, can you check `logs/pytest.log` to see the progress? The Socket Mode interactions tests tend to take much longer than others.

Reply:
It may be that... I do see connection reset by peer (also Broken Pipe error). Full log [here](https://gist.github.com/filmaj/a6de694e5bd6cd63917147b045c123e5).

As for the stuck `test_interactions_websocket_client.py` test, the `logs/pytest.log` seems to repeat a ConnectionRefusedError forever (full log [here](https://gist.github.com/filmaj/ddf54558348742a9f486a2034e7b9dc8))

Reply:
@filmaj I've sent a pull request fixing this issue. Thanks for pointing it out! https://github.com/slackapi/python-slack-sdk/pull/1072